### PR TITLE
 Improve plugin class and align it with new architecture - Closes #6623 

### LINKED ISSUE #6623: Improve plugin class and align it with new architecture
#### Init
To be consistent with the module, plugins should also use the “init” function instead of constructor.
```ts
init(context: PluginInitContext): Promise<void>;
interface PluginInitContext {
  config: Record<string, unknown>; // plugin config
  channel: PluginChannel;
  options: {
    dataPath: string;
    applicationConfig: ApplicationConfig; // Omit<ApplicationConfig, ‘plugins’> should not have plugin related property
  },
}
```
#### name

Rename the `alias` to `name` public property to be consistent with the `module`. Content should be like `httpAPI`.

```ts
public abstract name: string;
```

#### info

Remove info property and use `name` property instead wherever required. 
`exportPath` is needed if a plugin is expected to run in child process and must be implemented in the extended class. Rename to `nodeModulePath` as a function defined as below.
```ts
public abstract get nodeModulePath(): string;
```

#### default

Rename it to `configSchema`, or make it consistent with the module.

#### Update Tests

Update plugin related tests to adapt to prop changes and config input changes.

### Acceptance Criteria

- All the tests under `framework` and `framework-plugin` should pass

### Additional Information

- Related #6612

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #6623

### How was it solved?

Changes made as per the issue

### How was it tested?

All necessary tests updated

### CODE REVIEW

File: framework/src/controller/controller.ts
Comment (shuse2) [inline framework/src/controller/controller.ts:178]: This should use `plugin.name`

File: framework/src/controller/controller.ts
Comment (shuse2) [inline framework/src/controller/controller.ts:211]: same as above

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: framework/src/plugins/base_plugin.ts
Comment (Incede) [reply framework/src/plugins/base_plugin.ts:187]: I think this can't be done because creates a dependency cycle due to `BaseChannel` via bus if moved to types.ts

File: framework/src/application.ts
Comment (Incede) [reply framework/src/application.ts:185]: https://github.com/LiskHQ/lisk-sdk/blob/3b304af2b2323d3c786c642a9a91fe8931f99155/framework/test/unit/application.spec.ts#L531
In some test name is being passed through options, remove those test too ?

File: framework/src/plugins/base_plugin.ts
Comment (Incede) [reply framework/src/plugins/base_plugin.ts:300]: https://github.com/LiskHQ/lisk-sdk/blob/development/framework/test/unit/plugins/base_plugin.spec.ts#L146
I think this test is based on this try-catch block, and fails upon it's removal.

File: framework/src/application.ts
Comment (shuse2) [reply framework/src/application.ts]: yea sounds good! (as we dont use alias anymore

File: framework/src/plugins/base_plugin.ts
Comment (shuse2) [reply framework/src/plugins/base_plugin.ts]: Previously `info.name` was  the package name. That's why `require` with package name make sense. Now, the below block take care both patterns, so this is not necessary. test also can be removed as well

**Review by shuse2 — COMMENTED** [review]
LGTM in general!

File: framework/src/plugins/base_plugin.ts
Comment (nazarhussain) [inline framework/src/plugins/base_plugin.ts:195]: Since we made plugin `config` and `appConfig` two separate things, so this generic type needed to be changed. It should just be plugin config. If we want to really save `options` at plugin instance level then we create another attribute in the base plugin for it.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: framework/src/plugins/base_plugin.ts
Comment (nazarhussain) [inline framework/src/plugins/base_plugin.ts:197]: As said earlier, these options should be splitted as `config` and `options`

File: framework/src/plugins/base_plugin.ts
Comment (nazarhussain) [inline framework/src/plugins/base_plugin.ts:211]: It should not merge the config and options together.

File: framework/src/plugins/base_plugin.ts
Comment (nazarhussain) [inline framework/src/plugins/base_plugin.ts:272]:
```suggestion
	public get configSchema(): SchemaWithDefault | undefined {
```

File: framework/src/plugins/base_plugin.ts
Comment (nazarhussain) [inline framework/src/plugins/base_plugin.ts:286]: Please change this function name as well matching to plugin property you renamed to `nodeModulePath`

File: framework/src/plugins/base_plugin.ts
Comment (nazarhussain) [inline framework/src/plugins/base_plugin.ts:331]: In past we discussed to not use `assert` package any more. If that's the case we should refactor it now as we are already changing this file a lot. @shuse2

File: framework/src/plugins/base_plugin.ts
Comment (nazarhussain) [inline framework/src/plugins/base_plugin.ts:277]: I will suggest to remove these from `abstract` and not make these required for plugin developer to define. One can just develop a plugin without defining events and actions.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: framework/src/plugins/base_plugin.ts
Comment (shuse2) [reply framework/src/plugins/base_plugin.ts:331]: we don't use assert anymore, but since this PR is blocking other PR and it's already pretty big, refactoring later is also fine. up to @Incede  =)

File: framework/src/plugins/base_plugin.ts
Comment (shuse2) [reply framework/src/plugins/base_plugin.ts:277]: I will create issue to make properties to be consistent

**Review by nazarhussain — APPROVED** [review]
We decided to proceed with this PR but following points to be covered separately. 


1. Make plugin `name` a property rather the getter.
2. Rename function overridePluginOptions to updatePluginConfig  to be consistent across naming convention. 
3. Rename all files in plugins  `defaults/config.ts`  to` cofnigSchema.ts`  or `schema/config.ts`
4. Don’t need the code `objects.mergeDeep` in every plugin load  its already done in the base plugin. 
5. Need to remove `events` and `actions` from abstract members and create properties with default values. 
6. If possible create an issue to remove dependency of declaring `nodeModulePath` in every plugin.
7. Two functions `_loadInMemoryPlugin` and `_loadChildProcessPlugin` have extra parameter `name`. which can be removed now. Earlier it was the feature to register a plugin with different alias, now we restrict everywhere to just use plugin instance name. 
Remove the use of assert around new code case

**Comment by nazarhussain** [comment]
Created the issue to cover above points.

**Review by shuse2 — APPROVED** [review]

**Pull request merged**
